src/file.ts

```diff
@@ -2319,51 +2375,110 @@ class File extends ServiceObject<File> {
       config.action,
       config.contentMd5 || '',
       config.contentType || '',
-      expiresInSeconds,
+      config.expiration,
       extensionHeadersString + config.resource,
     ].join('\n');
 
     const authClient = this.storage.authClient;
-    authClient.sign(blobToSign)
-        .then(signature => {
-          authClient.getCredentials().then(credentials => {
-            const query = {
-              GoogleAccessId: credentials.client_email,
-              Expires: expiresInSeconds,
-              Signature: signature,
-            } as SignedUrlQuery;
-
-            if (typeof config.responseType === 'string') {
-              query['response-content-type'] = config.responseType!;
-            }
+    return authClient.sign(blobToSign)
+        .then(
+            signature => authClient.getCredentials().then(
+                credentials => ({
+                  GoogleAccessId: credentials.client_email!,
+                  Expires: config.expiration.toString(),
+                  Signature: signature,
+                })),
+            )
+        .catch((err) => {
+          throw new SigningError(err.message);
+        });
+  }
 
-            if (typeof config.promptSaveAs === 'string') {
-              query['response-content-disposition'] =
-                  'attachment; filename="' + config.promptSaveAs + '"';
-            }
-            if (typeof config.responseDisposition === 'string') {
-              query['response-content-disposition'] =
-                  config.responseDisposition!;
-            }
+  private getSignedUrlV4(config: GetSignedUrlConfigInternal):
+      Promise<SignedUrlQuery> {
+    const now = new Date();
+    const nowInSeconds = Math.floor(now.valueOf() / 1000);
```

Comment:
Can we prefer `Math.floor(Date.now() * 1000)`?

Reply:
would like to be able to re-use the same Date object, hence the same datetime throughout the method.